samples: matter: Reduce memory usage (#18554)
nrfconnect/sdk-nrf: doc/nrf/protocols/matter/getting_started/memory_optimization.rst

The static heap size means that you can define the maximum heap size for your application by setting the :kconfig:option:`CONFIG_CHIP_MALLOC_SYS_HEAP_SIZE` Kconfig value.
You can also adjust the heap dedicated for MbedTLS purposes by setting the :kconfig:option:`CONFIG_MBEDTLS_HEAP_SIZE` Kconfig option value.

> Packet Buffer Pool Size

Author: Fixed, thanks!

-----------------------

> The Matter application uses a fixed size pool for packet buffer memory management.

Reviewer: ```suggestion
The Matter application uses a fixed-size pool for packet buffer memory management.
```

The pool size is determined by the :kconfig:option:`CONFIG_CHIP_SYSTEM_PACKETBUFFER_POOL_SIZE` Kconfig option.

> You can adjust the pool size to reduce the memory usage of your application, however that may have a negative effect on the performance of the application.

Reviewer: ```suggestion
You can adjust the pool size to reduce the memory usage of your application, but that can have a negative effect on the performance of the application.
```

.. note::
   The Thread protocol has a separate configuration for memory allocation.

>    See :ref:`thread_configuring_messagepool` in the Thread documentation for more information.

Reviewer: Is it intentional that this note is under the new heading instead of the previous one?

Author: Yes, this note refers to configuration of pool used in Thread, so it fits better here.
